### PR TITLE
Fix upward merge Unit Tests

### PR DESCRIPTION
**Pull request by RamosFe**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/23839|

## Description

Fixes the errors related to the Unit Test of the wodles, framework, and api modules after merge between `4.9.0` and master.

### CODE REVIEW

File: framework/wazuh/core/wazuh_socket.py [hunk 1]
```diff
@@ -193,6 +193,7 @@ async def receive(self, header_format: str ="<I", header_size: int = 4) -> bytes
             raise WazuhException(1014, str(exc)) from exc
 
 
+
```
Comment (fdalmaup) [inline framework/wazuh/core/wazuh_socket.py:196]:
```suggestion
```

File: wodles/aws/tests/aws_utils.py [hunk 2]
```diff
@@ -251,7 +251,7 @@ def get_aws_bucket_parameters(db_table_name: str = TEST_TABLE_NAME, bucket: str
 
 
 def get_aws_service_parameters(db_table_name: str = TEST_TABLE_NAME, service_name: str = 'cloudwatchlogs',
-                               reparse: bool = False,
+                               reparse: bool = False, access_key: str = None, secret_key: str = None,
                                profile: str = TEST_AWS_PROFILE, iam_role_arn: str = None,
```
Comment (fdalmaup) [inline wodles/aws/tests/aws_utils.py:255]: This has been removed [here](https://github.com/wazuh/wazuh/pull/18680/files#diff-3e9d53faf228d4be2432c1e9b9da1b65d119c9c2c93f11960e2987181d2029a1) as part of the removal of the deprecated plain text auth.

File: framework/wazuh/core/tests/test_wazuh_socket.py [hunk 3]
```diff
@@ -2,42 +2,33 @@
...
-            self.on_data_received = ProtocolMock.testing()
-
-        @staticmethod
-        async def testing():
-            return 'test'
-        
-        @staticmethod
-        def get_data():
-            return b'data'
+@pytest.mark.asyncio
```
Comment (fdalmaup) [inline framework/wazuh/core/tests/test_wazuh_socket.py:24]:
```suggestion

@pytest.mark.asyncio
```

File: framework/wazuh/core/tests/test_wazuh_socket.py [hunk 4]
```diff
@@ -2,42 +2,33 @@
...
 import pytest
 from wazuh.core.exception import WazuhException
-from wazuh.core.wazuh_socket import WazuhSocket, WazuhSocketJSON, WazuhAsyncSocket, WazuhAsyncProtocol, \
-    SOCKET_COMMUNICATION_PROTOCOL_VERSION, create_wazuh_socket_message
+from wazuh.core.wazuh_socket import WazuhSocket, WazuhSocketJSON, \
+     SOCKET_COMMUNICATION_PROTOCOL_VERSION, create_wazuh_socket_message, WazuhAsyncSocket, \
+     WazuhAsyncSocketJSON
+
+@pytest.fixture
+def oux_conn_patch():
```
Comment (fdalmaup) [inline framework/wazuh/core/tests/test_wazuh_socket.py:16]: Did you mean `aux_conn_patch`?

**Review by fdalmaup — CHANGES_REQUESTED** [review]

File: framework/wazuh/core/tests/test_wazuh_socket.py (hunk 3 above)
Comment (RamosFe) [reply framework/wazuh/core/tests/test_wazuh_socket.py:25]: Fixed in commit 78e38c8d101c7ebce4a5b6820400cdcd54c2efd9 and documented results of tests in https://github.com/wazuh/wazuh/issues/23839#issuecomment-2161522591.

File: framework/wazuh/core/tests/test_wazuh_socket.py (hunk 4 above)
Comment (RamosFe) [reply framework/wazuh/core/tests/test_wazuh_socket.py]: Fixed in commit 78e38c8d101c7ebce4a5b6820400cdcd54c2efd9 and documented results of tests in https://github.com/wazuh/wazuh/issues/23839#issuecomment-2161522591.

File: wodles/aws/tests/aws_utils.py (hunk 2 above)
Comment (RamosFe) [reply wodles/aws/tests/aws_utils.py]: Fixed in commit e15342e3290b2ea66f47132b432a4b431e1fa4c4 and documented results of tests in https://github.com/wazuh/wazuh/issues/23839#issuecomment-2161536169.

**Review by fdalmaup — APPROVED** [review]
LGTM

File: tests/integration/test_api/test_config/test_logs_rotation/data/test_cases/cases_sizes_options.yaml
Comment (Selutario) [inline tests/integration/test_api/test_config/test_logs_rotation/data/test_cases/cases_sizes_options.yaml:1]: I finally decided to merge the fix for api conf error logs in 4.9.0:
- https://github.com/wazuh/wazuh/issues/24033

Leave this test_logs_rotation in a working state when the fix to `wazuh-apid.py` is upward merged to master (I guess that these changes are not needed in this case). The tests will fail until then, but we can justify them by pointing to #24033.

**Review by Selutario — CHANGES_REQUESTED** [review]

File: tests/integration/test_api/test_config/test_logs_rotation/data/test_cases/cases_sizes_options.yaml
Comment (RamosFe) [reply tests/integration/test_api/test_config/test_logs_rotation/data/test_cases/cases_sizes_options.yaml:1]: Rollback of changes in 4733a82c8dae2a96609467eb0189059c21bd31e5.

**Comment by RamosFe** [comment]
## Update

After running the AWS ITS, only tests related to `regions` failed https://github.com/wazuh/wazuh/actions/runs/9778882484/job/26996732508.

**Pull request merged**